src/battle/anim/wisp_fire.c

```diff
@@ -0,0 +1,48 @@
+#include "global.h"
+#include "battle_anim.h"
+#include "main.h"
+#include "rom_8077ABC.h"
+#include "trig.h"
+
+extern s16 gBattleAnimArgs[];
+extern u8 gAnimBankTarget;
+// extern u8 gAnimBankAttacker;
+
+void sub_80D5CC0(struct Sprite *sprite)
+{
+    if (!sprite->data[0])
+    {
+        sprite->data[1] = gBattleAnimArgs[0];
+        sprite->data[0] += 1;
+    }
+
+    sprite->data[3] += 0xC0 * 2;
+    sprite->data[4] += 0xA0;
+
+    sprite->pos2.x = Sin(sprite->data[1], sprite->data[3] >> 8);
+    sprite->pos2.y = Cos(sprite->data[1], sprite->data[4] >> 8);
+
+    sprite->data[1] = (sprite->data[1] + 7) & 0xFF;
+
+    if (gMain.inBattle)
+        if ((u16) (sprite->data[1] - 0x40) > 0x83)
```

Comment:
This is equivalent to `if (sprite->data[1] < 64 || sprite->data[1] > 195)`.

Reply:
Oh wow, totally didn't think about that, thanks!